src/main/java/gui/modal/AddCameraModalView.java

```diff
@@ -0,0 +1,126 @@
+package gui.modal;
+
+import java.util.ArrayList;
+
+import data.CameraType;
+import gui.events.AddCameraEvent;
+import gui.root.RootPane;
+import javafx.event.EventHandler;
+import javafx.scene.control.Button;
+import javafx.scene.control.Label;
+import javafx.scene.control.ListView;
+import javafx.scene.control.TextField;
+import javafx.scene.input.MouseEvent;
+import javafx.scene.layout.HBox;
+import javafx.scene.layout.VBox;
+import javafx.scene.text.Text;
+
+public class AddCameraModalView extends ModalView {
+    
+    private static final int width = 300;
+    private static final int height = 300;
+    
+    private TextField nameField;
+    private TextField descriptionField;
+    private ListView<Label> cameraTypes;
+    private Button addCameraButton;
+    private VBox viewPane;
+    private ArrayList<CameraType> cameraTypeList;
+    
+    private EventHandler<AddCameraEvent> eventHandler;
+    
+    public AddCameraModalView(RootPane rootPane,
+                              EventHandler<AddCameraEvent> handler,
+                              ArrayList<CameraType> types) {
+        this(rootPane, handler, types, width, height);
+    }
+    
+    /**
+     * Construct a new AddCameraModalView.
+     * @param rootPane the rootPane that uses this modal
+     * @param handler the handler to handle the result of the modal
+     * @param types the camera types that can be used
+     * @param width the width of the modal screen
+     * @param height the height of the modal screen
+     */
+    public AddCameraModalView(RootPane rootPane,
+                              EventHandler<AddCameraEvent> handler,
+                              ArrayList<CameraType> types, 
+                              int width, 
+                              int height) {
+        super(rootPane, width, height);
+        this.eventHandler = handler;
+        this.cameraTypeList = types;
+        initializeView();
+    }
+    
+    /**
+     * Initialize the view of this modal.
+     */
+    private void initializeView() {
+        this.viewPane = new VBox(20);
+        this.viewPane.getChildren().add(new Text("Create a camera"));
+        
+        initFields();
+        initTypeList();
+        
+        addCameraButton = new Button("Add Camera");
+        addCameraButton.setOnMouseClicked(this::addCamera);
```

Comment:
The idea of mvc is to add the handlers directly in the controllers to prevent overhead for unnecesary events. On the click of this button you make an event, fire it and collect it at the controller. If you just add the handler directly in the controller this overhed will be removed